iteration/iteration.go

```diff
@@ -187,15 +187,20 @@ func (m *GormIterationRepository) Root(ctx context.Context, spaceID uuid.UUID) (
 	var itr Iteration
 
 	tx := m.db.Where("space_id = ? and path = ?", spaceID, "").First(&itr)
-	if tx.Error != nil && tx.Error != gorm.ErrRecordNotFound {
-		log.Error(ctx, map[string]interface{}{
-			"space_id": spaceID,
-			"err":      tx.Error,
-		}, "unable to get the root iteration")
-		return nil, errors.NewInternalError(ctx, tx.Error)
+	if tx.Error == nil {
```

Comment:
I know this seems shorter but I guess in 99% of the code we have the logic for errors to be `if err != nil`. I wonder if this wouldn't work as well:

```go
if tx.Error != nil {
	log.Error(ctx, map[string]interface{}{
		"space_id": spaceID,
		"err":      tx.Error,
	}, "unable to get the root iteration")
        if tx.Error == gorm.ErrRecordNotFound {
		return nil, errors.NewNotFoundError("root iteration for space", spaceID.String())
	}
	return nil, errors.NewInternalError(ctx, tx.Error)
}
return &itr, nil
 ```

Notice that I've removed the switch and made the error message begin with lower case.

Reply:
I agree with @kwk, that's why I had initially implemented it the `if tx.Error != nil` way.
What do you think @tinakurian?

Reply:
@jarifibrahim i like the removal of the switch. i noticed that we do use error != nil in a lot (if not all) other places. In some places it actually does reduce nesting, but in this case it seemed like it could become more readable by doing the reverse. However, there isn't much nesting here to begin with so in either case, it looks good :-) :+1:

Reply:
Please see https://github.com/fabric8-services/fabric8-wit/pull/2126/commits/9f5f0bf515f6ee68d0c949cfbfbcac0b1d3f939f